### PR TITLE
[Backport 2.13] Added static setting for checkPendingFlushUpdate functionality of lucene index writer

### LINKED ISSUE #11628: [Performance Idea] Optimize Indexing tail latencies by disabling checkPendingFlushUpdate flag on DocumentWriter
### Is your feature request related to a problem? Please describe

Opensearch triggers a full flush on lucene based on the translog size or maybe if a refresh is triggered(sometimes) or flush API is called.

Sometimes indexing threads end up flushing the segments to the disk when a full flush has been triggered. This incurs latency overhead and reduces the throughput keeping the number of clients same.

Looking deeper, it appears that indexing threads check for pending flushes on update in order to help flushing indexing buffers to disk. We can turn this off by disabling [checkPendingFlushUpdate](https://lucene.apache.org/core/7_4_0/core/org/apache/lucene/index/LiveIndexWriterConfig.html#setCheckPendingFlushUpdate-boolean-) parameter in index config.

Also [IndexingMemoryController](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/indices/IndexingMemoryController.java) reduces the total number of indexing threads per shard to 1, in order to throttle indexing when it breaches 1.5 times of indexing buffer. It starts limiting the indexing rate starting with the shard with largest memory accounting.

### Describe the solution you'd like

We should ideally keep [checkPendingFlushUpdate](https://lucene.apache.org/core/7_4_0/core/org/apache/lucene/index/LiveIndexWriterConfig.html#setCheckPendingFlushUpdate-boolean-) disable but we do want the indexing threads to start help with flushing somewhere between when flushing starts and throttling starts to restore to old behaviour when flushing is slow.
checkForPendingFlush is a [LiveIndexWriterConfig](https://lucene.apache.org/core/7_3_0/core/org/apache/lucene/index/LiveIndexWriterConfig.html), which means that we can modify it on a live index writer instance. We can follow the existing procedure for enabling this flag used for throttling in IndexingMemoryController- start with the shard that has largest memory accounting and disable it once the memory is below the threshold.

### Related component

Indexing:Performance

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by CaptainDredge**
### Description
Backports PR #12786 to `2.13`

### Related Issues
Resolves #11628

### Check List
- [ ] ~New functionality includes testing.~
  - [ ] ~All tests pass~
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/pull/6728)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@CaptainDredge do we have a sign-off from @bbarani @gaiksaya to bring this one to `2.13`? (the release is cut already)

**Comment by CaptainDredge** [comment]
@reta I don't have sign-off yet but I'm wondering why this didn't get to `2.13` in the first place automatically when it was merged in `2.x` 2 days back i.e. much before RC release 😕

**Comment by reta** [comment]
> @reta I don't have sign-off yet but I'm wondering why this didn't get to `2.13` in the first place automatically when it was merged in `2.x` 2 days back i.e. much before RC release 😕

We are where we are :(, please ask for sign off

**Review by reta — CHANGES_REQUESTED** [review]
Blocking the merge till we get 2.13 clearance

**Comment by gaiksaya** [comment]
> @CaptainDredge do we have a sign-off from @bbarani @gaiksaya to bring this one to `2.13`? (the release is cut already)

Please go ahead.

**Comment by gaiksaya** [comment]
> @reta I don't have sign-off yet but I'm wondering why this didn't get to `2.13` in the first place automatically when it was merged in `2.x` 2 days back i.e. much before RC release 😕

2.13 was cut 3 days back honoring code cut off date. Anything merged after that needs to go in 2.13 as well.

**Comment by bbarani** [comment]
> Blocking the merge till we get 2.13 clearance

Thanks @reta for surfacing this issue to release team. @CaptainDredge Can you check and let us know why we missed this backport? Do we need to add any additional guard rails? We need to make sure this doesn't happen again for anyone.

**Comment by CaptainDredge** [comment]
@reta / @shwetathareja can you please merge the PR now since we have the signoff?

**Review by reta — APPROVED** [review]

File: CHANGELOG.md
Comment (reta) [inline CHANGELOG.md:33]: @CaptainDredge this has to go to `release-notes/opensearch.release-notes-2.13.0.md`

Comment (CaptainDredge) [reply CHANGELOG.md:33]: Done :)

**Review by reta — APPROVED** [review]

**Pull request merged**
